homeassistant/helpers/trigger_template_entity.py

```diff
@@ -176,35 +200,91 @@ def restore_attributes(self, last_state: State) -> None:
                 extra_state_attributes[attr] = last_state.attributes[attr]
             self._rendered[CONF_ATTRIBUTES] = extra_state_attributes
 
-    def _render_templates(self, variables: dict[str, Any]) -> None:
-        """Render templates."""
+    def _render_single_template(
+        self,
+        key: str,
+        variables: dict[str, Any],
+        strict: bool = False,
+    ) -> Any:
+        """Render a single template."""
         try:
-            rendered = dict(self._static_rendered)
+            if key in self._to_render_complex:
+                return render_complex(self._config[key], variables)
 
-            for key in self._to_render_simple:
-                rendered[key] = self._config[key].async_render(
-                    variables,
-                    parse_result=key in self._parse_result,
-                )
+            return self._config[key].async_render(
+                variables, parse_result=key in self._parse_result, strict=strict
+            )
+        except TemplateError as err:
+            log_triggered_template_error(self.entity_id, err, key=key)
 
-            for key in self._to_render_complex:
-                rendered[key] = render_complex(
-                    self._config[key],
-                    variables,
-                )
+        return _SENTINEL
 
-            if CONF_ATTRIBUTES in self._config:
-                rendered[CONF_ATTRIBUTES] = render_complex(
-                    self._config[CONF_ATTRIBUTES],
-                    variables,
+    def _render_templates(self, variables: dict[str, Any]) -> None:
+        """Render templates."""
+        rendered = dict(self._static_rendered)
+
+        # Check availability first and render as a simple template because
+        # availability should only be able to render True, False, or None.
+        available = True
+        if CONF_AVAILABILITY in self._to_render_simple:
```

Comment:
Maybe we should not store the availability template in `_to_render_simple`, but instead store a bool flag `_has_availability_template` which is set if the availability template is not static? Then we can avoid the search here and also avoid the continue in the loop further down.

Likewise, we could store the key of the state or value template.

Reply:
Doesn't matter to me.  That could be done pretty easily